src/js/server/hcapi/applications.js

```diff
@@ -1,37 +1,164 @@
+  let teamMembersScores = teamMembers.map((member) => {
+    let memberApplicationId = member.hacker.hackerApplication.id;
+    let score = individualScores[memberApplicationId];
+    return score ? score : null;
+  });
+  // Check that all the teamMembers have been scored
+  // Return the average of those scores
+  return averageOrNull(teamMembersScores);
+}
+
+// Takes a list of numbers and returns the average or null if any one of them is null
+function averageOrNull(values) {
+  let sum = 0;
+  let length = values.length;
+  for (var i = 0; i < length; i++) {
+    if (values[i] === null) { return null }
+    sum += values[i];
+  }
+  return sum / length;
+}
+
 applicationsRouter.get('/', (req, res, next) => {
-  HackerApplication
-    .findAll({
-      include: [
-        {
-          model: Hacker,
-          required: true,
-        },
-        {
-          model: ApplicationResponse,
-          required: false,
-        },
-      ],
-    })
-    .then(applications => applications.map(appl => ({
-      id: appl.id,
-      name: `${appl.hacker.firstName} ${appl.hacker.lastName}`,
-      gender: appl.hacker.gender,
-      country: appl.countryTravellingFrom,
-      inTeam: appl.inTeam,
-      rating: 0, // Temporary — will be replaced with actual score soon
-      status: appl.applicationResponse !== null ? (appl.applicationResponse === 'invited' ? 'Invited' : 'Not Invited') : 'Pending'
-    })))
-    .then(applications => {
-      res.json({
-        applications,
-      });
-    })
+
+  // Get the applications
+  let applicationsPromise = HackerApplication.findAll({
+    include: [
+      {
+        model: Hacker,
+        required: true,
+        include: [
+          {
+            model: TeamMember,
+            as: 'Team',
+            required: false,
+          },
+        ],
+      },
+      {
+        model: ApplicationResponse,
+        required: false,
+      },
+    ],
+  });
+
+  // Get the individual scores (exist if reviews have been done)
+  let individualScoresPromise = db.query(individualScoreQuery, {
+    type: Sequelize.QueryTypes.SELECT,
+  });
+
+  // Get the teams with members listed
+  let teamsPromise = Team.findAll({
+    include: [
+      {
+        model: TeamMember,
+        required: true,
+        include: [
+          {
+            model: Hacker,
+            include: [HackerApplication],
+          },
+        ],
+      }
+    ]
+  });
+
+  Promise.all([
+    applicationsPromise,
+    individualScoresPromise,
+    teamsPromise,
+  ]).then(([applications, individualScoresArr, teamsArr]) => {
+   
+    /**
+     * Flatten the scores array for later efficiency
+     * [
+     *   {application_id: 1, avg: 2.5},
+     *   {application_id: 2, avg: 3.5},
+     * ]
+     *
+     * is flattened to:
+     *
+     * {
+     *   '1': 2.5,
+     *   '2': 3.5,
+     * }
+     */
+    const individualScores = individualScoresArr.reduce((a, b) => Object.assign(a, {[b.application_id]: parseFloat(b.avg)}), {});
```

Comment:
If we're looking for efficiency here, an [ES6 map](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Map) may be more appropriate (but less functional so I'm torn)